### PR TITLE
CLOUDP-261495: Bump CLI to Atlas Operator 2.4.0

### PR DESCRIPTION
**Pull request by josvazg**
## Proposed changes

_Jira ticket:_ CLOUDP-#

Closes #[issue number]

## Checklist

- [X] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by s-urbaniak — CHANGES_REQUESTED** [review]
This is not enough, we also need to bump `LatestOperatorMajorVersion` and the `versionsToResourcesMap` which adds backup compliance policies.

**Comment by s-urbaniak** [comment]
See https://github.com/mongodb/mongodb-atlas-cli/blob/master/internal/kubernetes/operator/features/crds.go#L29-L86

File: internal/kubernetes/operator/features/crds.go [hunk 1]
```diff
@@ -83,6 +83,18 @@ var (
...
+			resource{ResourceAtlasDatabaseUser, NopPatcher()},
+			resource{ResourceAtlasProject, NopPatcher()},
+			resource{ResourceAtlasDeployment, NopPatcher()},
+			resource{ResourceAtlasBackupSchedule, NopPatcher()},
+			resource{ResourceAtlasBackupPolicy, NopPatcher()},
+			resource{ResourceAtlasTeam, NopPatcher()},
+			resource{ResourceAtlasDataFederation, NopPatcher()},
+			resource{ResourceAtlasFederatedAuth, NopPatcher()},
+			resource{ResourceAtlasStreamInstance, NopPatcher()},
+			resource{ResourceAtlasStreamConnection, NopPatcher()},
```
Comment (s-urbaniak) [inline internal/kubernetes/operator/features/crds.go:96]: this is missing the backup compliance policy resources

Comment (igor-karpukhin) [reply internal/kubernetes/operator/features/crds.go:96]: Yes, this one has to be added

**Review by igor-karpukhin — CHANGES_REQUESTED** [review]
BCP crd is missing in the AKO resources list

**Review by s-urbaniak — APPROVED** [review]
lgtm if the e2e kubernetes evergreen test is green

File: internal/kubernetes/operator/features/crds.go (hunk 1 above)
Comment (josvazg) [reply internal/kubernetes/operator/features/crds.go:97]: done

File: internal/kubernetes/operator/features/crds.go [hunk 2]
```diff
@@ -83,6 +84,19 @@ var (
 			resource{ResourceAtlasStreamInstance, NopPatcher()},
 			resource{ResourceAtlasStreamConnection, NopPatcher()},
 		},
+		"2.4.0": {
```
Comment (helderjs) [inline internal/kubernetes/operator/features/crds.go:87]: You should remove the 2.1.0 entry. Unless something has changed we only support the 3 latest (major) versions of AKO

**Review by helderjs — CHANGES_REQUESTED** [review]

File: internal/kubernetes/operator/features/crds.go (hunk 2 above)
Comment (josvazg) [reply internal/kubernetes/operator/features/crds.go:77]: Done

**Comment by helderjs** [comment]
@josvazg You need to fix the test https://github.com/mongodb/mongodb-atlas-cli/blob/3bfd384a54d6f36842786e67c2acd790e6e31d0b/test/e2e/atlas/kubernetes_config_generate_test.go#L888 and line 839

but I think this PR is not need until we implement BCP export

**Review by s-urbaniak — APPROVED** [review]

**Review by igor-karpukhin — APPROVED** [review]

**Review by helderjs — APPROVED** [review]

**Pull request merged**